intranet/static/js/common.js

```diff
@@ -31,25 +30,162 @@ $(function() {
+                    })
+                },
+                check_direction: function(link) {
+                    x_magnitude = Math.abs(this.start_x - this.stop_x);
+                    y_magnitude = Math.abs(this.start_y - this.stop_y);
+                    x = ((this.start_x - this.stop_x) < 0) ? "RIGHT" : "LEFT";
+                    y = ((this.start_y - this.stop_y) < 0) ? "DOWN" : "UP";
+                    result = (x_magnitude > y_magnitude) ? x : y;
+                    result = (this.tap == true) ? "TAP" : result;
+                    if (result == this.keys[0]) this.keys = this.keys.slice(1, this.keys.length);
+                    if (this.keys.length == 0) {
+                        this.keys = this.orig_keys;
+                        this.code(link);
+                    }
+                }
+            }
+        };
+        typeof callback === "string" && udlr.load(callback);
+        if (typeof callback === "function") {
+            udlr.code = callback;
+            udlr.load();
+        };
+        return udlr
+    })(window.creffettMode = function() {
+        $("body").addClass("fire").click(function() {
+            $("iframe#udlr").remove();
+            $("body").removeClass('fire');
+        }).append('<iframe id="udlr" style="position:fixed;top:50%;left:50%;width:640px;height:480px;margin:-240px -320px" width="640" height="480" src="https://www.youtube.com/embed/kfATGWeWqgc?autoplay=1&loop=0" frameborder="0" allowfullscreen></iframe>')
+    });
+    $(function() {
+        if (location.search.indexOf('creffett=1') !== -1) creffettMode()
+    });
+} catch (e) {}
+
+runEgg = function(q) {
+    switch (q) {
+        case "do a barrel roll":
+            setTimeout(function() { ytwin("mv5qzMtLE60") }, 1);
+            setTimeout(function() {
+                $("body").append("<style>@-webkit-keyframes roll { from { -webkit-transform: rotate(0deg) } to { -webkit-transform: rotate(360deg) } } @-moz-keyframes roll { from { -moz-transform: rotate(0deg) } to { -moz-transform: rotate(360deg) } } @keyframes roll { from { transform: rotate(0deg) } to { transform: rotate(360deg) } } body {-moz-animation-duration: 4s;-moz-animation-iteration-count: 4;-moz-animation-name: roll;-webkit-animation-name: roll; -webkit-animation-duration: 4s; -webkit-animation-iteration-count: 4;animation-name: animation-duration: 4s; animation-iteration-count: 4;}</style>");
+            }, 2000);
+            break;
+        case "asteroids":
+            var KICKASSVERSION = '2.0';
+            var s = document.createElement('script');
+            s.type = 'text/javascript';
+            document.body.appendChild(s);
+            s.src = '//hi.kickassapp.com/kickass.js';
+            $("body").append("<style>div#kickass-menu.KICKASSELEMENT {display: none !important;top: -9999px;left: -9999px;}</style>");
+            break;
+        case "d is for dogecoin":
+            var s = document.createElement("script");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is already declared in line 163.](https://www.codacy.com/app/tjhsst-ion/ion/file/2769892630/issues/source?bid=3212528&fileBranchId=3355681#l170)